pkg/controller/projects/controller.go

```diff
@@ -337,10 +342,15 @@ func generateProjectObservation(r *argocdv1alpha1.AppProject) v1alpha1.ProjectOb
 func generateCreateProjectOptions(p *v1alpha1.Project) *project.ProjectCreateRequest {
 	projSpec := generateProjectSpec(&p.Spec.ForProvider)
 
+	projName := meta.GetExternalName(p)
+	if projName == "" {
```

Comment:
Has this been tested with existing resources? I am wondering since a lot of crossplane controllers have this `NewNameAsExternalName` initializer so we might not necessarily need to have this extra check here.